Utils/Dataflow/017_adjustMetadata/adjustMetadata.py

```diff
@@ -293,6 +294,65 @@ def transform_chain_data(data):
     return True
 
 
+def ami_tags_chain(data):
+    """ Get AMI tags chain from task name.
+
+    :param data: task metadata
+    :type data: dict
+
+    :returns: AMI tags chain,
+              None if input data do not provide enough information
+    :rtype: str, NoneType
+    """
+    taskname = data.get('taskname')
+    try:
+        ami_tags = taskname.split('.')[-1]
+    except AttributeError:
+        # `taskname` is None or something else, but not a string
+        ami_tags = None
+    return ami_tags
+
+
+def generate_step_names(data):
+    """ Add fields with name of step to which task belongs.
+
+    There are different ways to say one step from another:
+    - MC production step name (already exists as 'step_name' field);
+    - current AMI tag + output data format;
+    - chain of AMI tags + output data format.
+
+    The latter is supposed to be the most universal, but initially was used
+    only the first one, then for some cases the second was invened, and...
+    ...and there's no good way to make things as they are supposed to be
+    all at once. So we need all the possible namings.
+
+    :param data: task metadata (will be altered in place)
+    :type data: dict
+
+    :returns: None
+    :rtype: NoneType
+    """
+    ignore_formats = ['LOG']
```

Comment:
`PHYS_CATEGORIES_HASHTAGS_MAP` is really massive and having it in the function makes things much less readable. Also, it is kind of a definition for "category" -- in other words, it fully defines how we map hashtags to categories. And it does not seem to requre much more context.

`ignore_formats` is not that massive, and much much more context-dependent. I mean, how should I name it, if it's a global variable? `IGNORE_FORMATS_FOR_STEPS_DEFINITION`? Still not very clear, what it is... it could be `TECHNICAL_FORMATS`, but as for now -- it is used only for steps definition, and for other use-cases "technical" might mean something else.

So I'd rather not.